.github/workflows/cicd.yml

```diff
@@ -0,0 +1,448 @@
+      
+      aws_region: ${{ steps.aws.outputs.aws_region }}
+      aws_s3: ${{ steps.aws.outputs.aws_s3 }}
+      
+      do_create_test_cluster: ${{ steps.aws.outputs.do_create_test_cluster }}
+
+    steps:
+      - name: git
+        id: git
+        run: |
+          echo $GITHUB_REF
+          git_branch=$(echo ${GITHUB_REF##*/})
+          
+          echo "::set-output name=git_branch::$git_branch"
+          
+          # TODO: Change after merge !!
+          if [[ $git_branch == "infrastructure-as-code" ]]
+          then
+            echo "::set-output name=do_deploy_to_prod_cluster::true"
+          else
+            echo "::set-output name=do_deploy_to_prod_cluster::false"
+          fi
+          
+      - name: checkout
+        uses: actions/checkout@v2
+          
+      - name: files
+        id: files
+        uses: jitterbit/get-changed-files@v1
+        
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: versions
+        id: versions
+        run: |
+          rasa_version=$( cat requirements.txt | grep 'rasa\[spacy\]' | cut -d'=' -f 3 )-spacy-en
+          rasa_sdk_version=$( cat requirements.txt | grep 'rasa-sdk' | cut -d'=' -f 3 )
+          
+          echo "::set-output name=rasa_enterprise_version::${{ env.RASA_ENTERPRISE_VERSION }}"
+          echo "::set-output name=rasa_version::$rasa_version"
+          echo "::set-output name=rasa_sdk_version::$rasa_sdk_version"
+        
+      - name: action_server
+        id: action_server
+        run: |
+          if [[ "${{ steps.files.outputs.all }}" == *"actions/"* ]]
+          then
+            echo "::set-output name=do_action_server::true"
+          else
+            echo "::set-output name=do_action_server::false"
+          fi
+
+      - name: rasa_model
+        id: rasa_model
+        run: |
+          if [[ "${{ steps.files.outputs.all }}" == *"data/"* ]] || \
+             [[ "${{ steps.files.outputs.all }}" == *"config.yml"* ]] || \
+             [[ "${{ steps.files.outputs.all }}" == *"domain.yml"* ]]
+          then
+            echo "::set-output name=do_training::true"
+          else
+            echo "::set-output name=do_training::false"
+          fi
+      
+      - name: aws
+        id: aws
+        run: |
+          echo "::set-output name=aws_region::${{ env.AWS_REGION }}"
+          echo "::set-output name=aws_s3::${{ env.AWS_S3_BUCKET_NAME }}"
+              
+          if [[ $(make aws-eks-cluster-exists) = True ]]
+          then
+            echo "::set-output name=do_create_test_cluster::false"
+          else
+            echo "::set-output name=do_create_test_cluster::true"
+          fi
+      
+
+        
+  params_summary:
```

Comment:
Can params_summary be folded in as a step in params?

Reply:
It could, but I like it better as a separate job, because of the following reasons:

- It provides a good test that the params can indeed be used by downstream jobs.
- It provides an example how to use the params in downstream jobs.
- It is easier to go look at the parameters. Just one click on the `params_summary` job, instead of having to click on the `params` jobs, and then on a `params_summary` step.